### PR TITLE
Implement initial styling of navigation menu screen

### LINKED ISSUE #21284: Style the Navigation page.
**Is your feature request related to a problem? Please describe.**
#21036 implemented the nav page. However the styling is very basic currently and needs to be improved.

**Describe the solution you'd like**
Designs for this are being explored in #19170:

v4

v4 1

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #21284

I've implemented this using `@wordpress/components` components, some of which look different from #21284, just because of the styles of the components.

We'd have to decide if we want to create variations of the styles those components offer or just stick with the out of the box styles.

There's also a question over whether some of the layout makes semantic sense. E.g. a Save Button in a Panel title.

These things can be iterated on. 

## How has this been tested?
1. Visit the Navigation Menu page.
2. Observe the styled page.

## Screenshots 
Screenshot 2020-04-01 at 4 48 25 pm

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by mtias — APPROVED** [review]

**Comment by noisysocks** [comment]
> I've implemented this using `@wordpress/components` components, some of which look different from #21284, just because of the styles of the components.

I personally think this is the right thing to do. Over time I'd like to see `@wordpress/components` become the canonical look and feel of WP Admin.

File: packages/edit-navigation/src/components/menu-editor/index.js [hunk 1]
```diff
@@ -35,24 +33,35 @@ export default function MenuEditor( { menuId, blockEditorSettings } ) {
...
+								showNestedBlocks
+								showAppender
+							/>
+						) }
+					</PanelBody>
+				</Panel>
+				<Panel
+					header={
+						<Button isPrimary onClick={ saveBlocks }>
+							{ __( 'Save Navigation' ) }
```
Comment (noisysocks) [inline packages/edit-navigation/src/components/menu-editor/index.js:52]: Let's use sentence casing here (see https://github.com/WordPress/gutenberg/pull/18758).

```suggestion
							{ __( 'Save navigation' ) }
```

File: packages/edit-navigation/src/components/menus-editor/index.js [hunk 2]
```diff
@@ -28,14 +28,21 @@ export default function MenusEditor( { blockEditorSettings } ) {
...
 		<>
-			<SelectControl
-				label={ __( 'Select navigation to edit:' ) }
-				options={ menus.map( ( menu ) => ( {
-					value: menu.id,
-					label: menu.name,
-				} ) ) }
-				onChange={ ( selectedMenuId ) => setMenuId( selectedMenuId ) }
-			/>
+			<Card className="edit-navigation-menus-editor__menu-selection-card">
```
Comment (noisysocks) [inline packages/edit-navigation/src/components/menus-editor/index.js:31]: A `Card`! Very cool!

**Review by noisysocks — APPROVED** [review]
This looks great! Nice work. It's really cool to see how adaptable `@wordpress/components` are now.

We'll need to make this responsive for mobile and tablet viewports but we can do that in a follow-up PR since this is all marked experimental.

**Comment by noisysocks** [comment]
Let's get this in and iterate. It's a huge improvement that brings us closer to being able to test this new screen on users.

I created https://github.com/WordPress/gutenberg/issues/21337 to track making this screen responsive.

@karmatosed: If there's any design feedback here, plop it into [the project board](https://github.com/WordPress/gutenberg/projects/31) and we'll get it sorted.

**Pull request merged**